### ISSUE TITLE
precisions on training / testing

### ISSUE THREAD

**Issue by Nal44** [Kaiseem/AD-GAN#1]
Hi,

I found your paper on bioarxiv, this is what I was looking for !
Impressed by the result of the gan :) .

I want to test the algorithm on my images I a few questions to understand how it works ;

For the training :
Do I need images and grand truth or ONLY images ?
Does the training .py only use the training images OR both training and test images+ground truth ? 

For the testing : 
does it only use the testing images+ground truth ? 

Then the trainer generate both fake images and grand truth ? 
and get the metrics based on the test set ? 

I have started to put everything in a Jupyter notebook for ease, but these info will help me to understand the structure better.

Thanks a lot!

**Comment by Kaiseem** [comment]
Hi,

For training, you don't need ground truth masks. Instead, you need to manually estimate the number and the size of the nuclei, then synthesize the generated masks, as described in instruction 2. Then, if the estimated properties of nuclei is appropriate, you can train the model with ONLY images.

For testing, you need both images and ground truth, so that you can figure out if the performance of the trained model aligns with human annotations. 

In addition, as a GAN, you can also input the synthesized masks to generate fake images, then you can have paired images and masks for downstream tasks.

Hoping my answer would help you :)

**Comment by Nal44** [comment]
Hi,
Thanks for the clarifications,

Just to make sure for training : 
I can use either the images , OR the  synthesized masks , correct ? 

Testing : paired images.

Once I have that I can send the Jupiter notebook that I am making if interested :) .

Cheers from the US,
antho

**Comment by Kaiseem** [comment]
Hi,

For  simplification, 

Training require:
Images and some hyperparameters as follow:
```bash
python train.py --name=ADGAN --dataroot=datasets/YourDATA --ellipse_min_radius=XX --ellipse_max_radius=XX --ellipse_min_num=XX --ellipse_max_num=XX --dimension=2
```
the hyperparameters are integers you need to specify, e.g., ellipse_min_radius, ellipse_max_num

Testing require:
paired images and corresponding masks

:)

**Comment by Nal44** [comment]
Hi,

Ok , yes I got that.
To use the GAN though how does it work ?
I am quite interested to test both :) .

Thanks a lot,
